test/framework/src/main/java/org/elasticsearch/test/fixture/AbstractHttpFixture.java

```diff
@@ -75,85 +80,100 @@ public final void listen() throws IOException, InterruptedException {
-                            response = handle(request);
+                        final Request request = new Request(requestId, method, exchange.getRequestURI(), headers, body.toByteArray());
+                        response = handle(request);
 
-                        } catch (Exception e) {
-                            final String error = e.getMessage() != null ? e.getMessage() : "Exception when processing the request";
-                            response = new Response(500, singletonMap("Content-Type", "text/plain; charset=utf-8"), error.getBytes(UTF_8));
-                        }
+                    } catch (Exception e) {
+                        final String error = e.getMessage() != null ? e.getMessage() : "Exception when processing the request";
+                        response = new Response(500, singletonMap("Content-Type", "text/plain; charset=utf-8"), error.getBytes(UTF_8));
                     }
+                }
 
-                    if (response == null) {
-                        response = new Response(400, TEXT_PLAIN_CONTENT_TYPE, EMPTY_BYTE);
-                    }
+                if (response == null) {
+                    response = new Response(400, TEXT_PLAIN_CONTENT_TYPE, EMPTY_BYTE);
+                }
 
-                    response.headers.forEach((k, v) -> exchange.getResponseHeaders().put(k, singletonList(v)));
-                    if (response.body.length > 0) {
-                        exchange.sendResponseHeaders(response.status, response.body.length);
-                        exchange.getResponseBody().write(response.body);
-                    } else {
-                        exchange.sendResponseHeaders(response.status, -1);
-                    }
-                } finally {
-                    exchange.close();
+                response.headers.forEach((k, v) -> exchange.getResponseHeaders().put(k, singletonList(v)));
+                if (response.body.length > 0) {
+                    exchange.sendResponseHeaders(response.status, response.body.length);
+                    exchange.getResponseBody().write(response.body);
+                } else {
+                    exchange.sendResponseHeaders(response.status, -1);
                 }
-            });
-            httpServer.start();
+            } finally {
+                exchange.close();
+            }
+        });
+        httpServer.start();
+    }
 
-            // Wait to be killed
-            Thread.sleep(Long.MAX_VALUE);
+    protected abstract Response handle(Request request) throws IOException;
 
-        } finally {
+    protected void stop() {
+        if(httpServer != null) {
             httpServer.stop(0);
         }
     }
 
-    protected abstract Response handle(Request request) throws IOException;
+    public String getAddress() {
+        return "http://127.0.0.1:" + httpServer.getAddress().getPort();
```

Comment:
httpServer.getAddress().getHost() didn't work here for me as it returned 0.0.0.0.0.0. which isn't picked up by the fixture repo installation in the test class